### ISSUE TITLE
[Core\let] SIGSEGV error when using the alias `:`

### ISSUE THREAD

**Issue by RickBarretto** [arturo-lang/arturo#1126]
**Describe the bug**

```
$> let 'a
>> Runtime | cannot perform let
     error | not enough parameters: 2 required
$> a:
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
```

`a:` without arguments should return a Runtime error as for `let 'a` without arguments.

**Desktop (please complete the following information):**
 - arch:  `amd64/windows`
 - version: `v/0.9.83 b/331 @ 2023-03-30T15:09:55-03:00`

**Additional context**
[This message from Discord](https://discord.com/channels/765519132186640445/829324913097048065/1090773313590132776)

**Comment by drkameleon** [comment]
Tricky one!

Again the AST generation comes into play (since `a:` is not technically `let` yet...) So, apparently, it keeps looking for a right-hand parameter that doesn't exist... Let's see!...

Good catch and an important issue that we should somehow tackle! 🚀

**Comment by RickBarretto** [comment]
> since a: is not technically let yet...

Yeah, I notice this since `let` just can get `:block` `:string` or `:literal`, I can't do `let a "some text"`, while I can do it with `:`: `a: "some text"`

**Comment by RickBarretto** [comment]
I just discovered a few more things. As, for example, this being a valid syntax: `: 'a 10`...

![image](https://github.com/arturo-lang/arturo/assets/78623871/89a784d0-1498-4a98-bae4-0d59aee56a4f)
